app/controllers/projects_controller.rb

```diff
@@ -89,4 +94,41 @@ def project_params
     end
     params['project']
   end
+  
+  private
+  def order_by_project_name
+    if session[:pre_click] == "project_name"
+      if session[:order] == "ASC"
+        @projects = Project.order(name: :desc)
+      else
+        @projects = Project.order(:name)
+      end
+    else    
+      @projects = Project.order(:name)  
+    end
+    change
+  end
+  
+  def order_by_metric_name 
+    click_type = params[:type]
+    if session[:pre_click] == click_type
+      if session[:order] == "ASC"
+        @projects = Project.joins(:metric_samples).where("metric_samples.metric_name = ?", click_type).order("metric_samples.score")
+      else
+        @projects = Project.joins(:metric_samples).where("metric_samples.metric_name = ?", click_type).order("metric_samples.score").reverse
+      end
+    else
+      @projects = Project.joins(:metric_samples).where("metric_samples.metric_name = ?", click_type).order("metric_samples.score")
+    end 
+    change
```

Comment:
I'm a little bit concerned about the logic of `change` here
suppose a feature test is like this
```
I click project_name --> order changed from ASC to DESC
then I click some metric name --> order changed from DESC back to ASC
then I click project_name again --> I am expecting to see projects order in DESC but now it's still ordered in ASC
```
pls correct me if I was wrong :)